File: Max Xor Subarray of size K.py
class Solution:
    def maxSubarrayXOR(self, arr, k):
        # code here
        xor=0
        ans=0
        for i in range(k-1):
            xor=xor^arr[i]
        for i in range(k-1,len(arr)):
            xor=xor^arr[i]
            if xor>ans:
                ans=xor
            xor=xor^arr[i-k+1]
        return ans

File: Max Xor Subarray of size K.cpp
class Solution {
public:
    int maxSubarrayXOR(vector<int>& arr, int k) {
        int ans = 0, x = 0;
        
        for (int i = 0; i < arr.size(); i++) {
            x ^= arr[i];
            if (i >= k) x ^= arr[i - k];
            if (i >= k - 1) ans = max(ans, x);
        }
        
        return ans;
    }
};

